src/Zio/FileSystems/FileSystem.cs

```diff
@@ -612,7 +612,12 @@ private void AssertNotDisposed()
     {
         if (IsDisposing || IsDisposed)
         {
-            throw new ObjectDisposedException($"This instance `{GetType()}` is already disposed.");
+            Throw(this.GetType());
+        }
+
+        static void Throw(Type type)
```

Comment:
Moving the trowing of the exception out of the method, allows the method to be inlined.